src/components/FormStep.js

```diff
@@ -395,7 +440,15 @@ const FormStep = ({
     // formio form not mounted -> nothing to do
     if (!formRef.current) return;
 
-    dispatch({type: 'BLOCK_SUBMISSION'});
+    // backend logic leads to changes in FormIO configuration, which triggers onFormIOInitialized.
+    // This in turn triggers the onFormIOChange event because the submission data is set
+    // programmatically. Without checking for human interaction, this would block the
+    // submission again for LOGIC_CHECK_DEBOUNCE ms, for the logic check to eventually
+    // be interrupted inside the evaluateFormLogic handler because the data hasn't
+    // changed. We can skip this particular block-unblock cycle by only blocking the
+    // submission because of human input.
+    modifiedByHuman && dispatch({type: 'BLOCK_SUBMISSION'});
```

Comment:
not a huge fan of this boolean construct over a simple visible if-statements

Reply:
changed!